testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/AbstractTestHiveViews.java

```diff
@@ -167,6 +168,42 @@ public void testSelectOnViewFromDifferentSchema()
         assertViewQuery("SELECT * FROM test_schema.hive_test_view_1", queryAssert -> queryAssert.hasRowsCount(25));
     }
 
+    @Test(groups = HIVE_VIEWS)
+    public void testViewReferencingTableInDifferentSchema()
+    {
+        String schemaX = "test_view_table_in_different_schema_x" + randomTableSuffix();
+        String schemaY = "test_view_table_in_different_schema_y" + randomTableSuffix();
+        String tableName = "test_table";
+        String viewName = "test_view";
+
+        onHive().executeQuery(format("CREATE SCHEMA %s", schemaX));
+
+        onHive().executeQuery(format("CREATE SCHEMA %s", schemaY));
+
+        onTrino().executeQuery(format("CREATE TABLE %s.%s AS SELECT * FROM tpch.tiny.nation", schemaY, tableName));
+
+        onHive().executeQuery(format("CREATE VIEW %s.%s AS SELECT * FROM %s.%s", schemaX, viewName, schemaY, tableName));
+
+        assertThat(onTrino().executeQuery(format("SELECT COUNT(*) FROM %s.%s", schemaX, viewName))).containsOnly(row(25));
+    }
+
+    @Test(groups = HIVE_VIEWS)
+    public void testViewReferencingTableInTheSameSchemeWithoutQualifier()
+    {
+        String schemaX = "test_view_table_same_schema_without_qualifier_schema" + randomTableSuffix();
+        String tableName = "test_table";
+        String viewName = "test_view";
+
+        onHive().executeQuery(format("DROP SCHEMA IF EXISTS %s CASCADE", schemaX));
```

Comment:
Is this statement actually necessary ?  The schema name contains the `randomTableSuffix()` .